File: data-pipeline/telemetry-extractor/src/main/java/org/ekstep/ep/samza/task/TelemetryExtractorConfig.java
package org.ekstep.ep.samza.task;

import org.apache.samza.config.Config;

public class TelemetryExtractorConfig {
	
	private final String JOB_NAME = "TelemetryExtractor";
	private final String errorTopic;
	private String successTopic;
	private String assessTopic;
	private String defaultChannel;
	private final int dupStore;
	private int expirySeconds;
	private int rawIndividualEventMaxSize;

	public TelemetryExtractorConfig(Config config) {
		successTopic = config.get("output.success.topic.name", "telemetry.raw");
		assessTopic = config.get("output.assess.topic.name", "telemetry.assess.redact");
		errorTopic = config.get("output.error.topic.name", "telemetry.extractor.failed");
		defaultChannel = config.get("default.channel", "01250894314817126443");
		dupStore = config.getInt("redis.database.duplicationstore.id", 1);
		expirySeconds = config.getInt("redis.database.key.expiry.seconds", 432000);
		rawIndividualEventMaxSize = config.getInt("raw.individual.event.maxsize", 996148);
	}

	public String successTopic() {
		return successTopic;
	}
	
	public String assessTopic() {
    return assessTopic;
  }

	public String errorTopic() {
		return errorTopic;
	}

	public String jobName() {
		return JOB_NAME;
	}

	public String defaultChannel() {
		return defaultChannel;
	}

	public int dupStore() {
		return dupStore;
	}

	public int expirySeconds() {
		return expirySeconds;
	}

	public int rawIndividualEventMaxSize() {
		return rawIndividualEventMaxSize;
	}
}

File: data-pipeline-flink/telemetry-extractor/src/main/scala/org/sunbird/dp/task/TelemetryExtractorConfig.scala
package org.sunbird.dp.task

import java.util

import org.apache.flink.api.common.typeinfo.TypeInformation
import org.apache.flink.api.java.typeutils.TypeExtractor
import org.apache.flink.streaming.api.scala.OutputTag
import com.typesafe.config.Config
import org.sunbird.dp.core.job.BaseJobConfig

class TelemetryExtractorConfig(override val config: Config) extends BaseJobConfig(config, "telemetry-extractor") {

  private val serialVersionUID = 2905979434303791379L

  implicit val mapTypeInfo: TypeInformation[util.Map[String, AnyRef]] = TypeExtractor.getForClass(classOf[util.Map[String, AnyRef]])
  implicit val stringTypeInfo: TypeInformation[String] = TypeExtractor.getForClass(classOf[String])

  val dedupStore: Int = config.getInt("redis.database.duplicationstore.id")
  val cacheExpirySeconds: Int = config.getInt("redis.database.key.expiry.seconds")

  // Kafka Topics Configuration
  val kafkaInputTopic: String = config.getString("kafka.input.topic")
  val kafkaSuccessTopic: String = config.getString("kafka.output.success.topic")
  val kafkaDuplicateTopic: String = config.getString("kafka.output.duplicate.topic")
  val kafkaFailedTopic: String = config.getString("kafka.output.failed.topic")
  val eventMaxSize: Long = config.getLong("kafka.event.max.size")

  val deDupParallelism: Int = config.getInt("task.dedup.parallelism")
  val extractionParallelism: Int = config.getInt("task.extraction.parallelism")
  
  val UNIQUE_EVENTS_OUTPUT_TAG = "unique-events"
  val RAW_EVENTS_OUTPUT_TAG = "raw-events"
  val FAILED_EVENTS_OUTPUT_TAG = "failed-events"
  val LOG_EVENTS_OUTPUT_TAG = "log-events"
  val DUPLICATE_EVENTS_OUTPUT_TAG = "duplicate-events"

  // Metric List
  val successEventCount = "success-event-count"
  val failedEventCount = "failed-event-count"
  val auditEventCount = "audit-event-count"
  val totalBatchEventCount = "batch-event-count"
  
  val rawEventsOutputTag: OutputTag[util.Map[String, AnyRef]] = OutputTag[util.Map[String, AnyRef]](RAW_EVENTS_OUTPUT_TAG)
  val failedEventsOutputTag: OutputTag[util.Map[String, AnyRef]] = OutputTag[util.Map[String, AnyRef]](FAILED_EVENTS_OUTPUT_TAG)
  val logEventsOutputTag: OutputTag[util.Map[String, AnyRef]] = OutputTag[util.Map[String, AnyRef]](LOG_EVENTS_OUTPUT_TAG)

  val duplicateEventOutputTag: OutputTag[util.Map[String, AnyRef]] = OutputTag[util.Map[String, AnyRef]](id = DUPLICATE_EVENTS_OUTPUT_TAG)
  val uniqueEventOutputTag: OutputTag[util.Map[String, AnyRef]] = OutputTag[util.Map[String, AnyRef]](id = UNIQUE_EVENTS_OUTPUT_TAG)

}
